core/src/main/scala/com/salesforce/op/stages/sparkwrappers/specific/OpPredictorWrapper.scala

```diff
@@ -98,15 +103,22 @@ class OpPredictorWrapper[E <: Predictor[Vector, E, M], M <: PredictionModel[Vect
       .set(po, getOutputFeatureName)
       .fit(dataset)
 
-    SparkModelConverter.toOP(model, uid)
+    val wrappedModel = SparkModelConverter.toOP(model, uid)
       .setParent(this)
       .setInput(in1.asFeatureLike[RealNN], in2.asFeatureLike[OPVector])
       .setMetadata(getMetadata())
       .setOutputFeatureName(getOutputFeatureName)
+
+    if (model.isInstanceOf[XGBoostClassificationModel] || model.isInstanceOf[XGBoostRegressionModel]) {
+      wrappedModel.setOutputDF(model.transform(dataset.limit(1)))
+    }
+
+    wrappedModel
   }
+
 }
 
-abstract class OpPredictorWrapperModel[M <: PredictionModel[Vector, M]]
+abstract class OpPredictorWrapperModel[M <: PredictionModel[Vector, M] : ClassTag]
```

Comment:
If you need the `ClassTag` here then please add it to the implicit arguments list and name it, e.g. `implicit ctag: ClassTag`.